### PR TITLE
JBPM-8391: Combined view of task comments and task work tabs

### PR DESCRIPTION
**Pull request by nmirasch**
* Added a configuration parameter to preferences to enable display comments at work tab.
* Updated form container to be adjusted to form height. 
* Changed comments design to be displayed as list of items instead a table

### CODE REVIEW

**Comment by nmirasch** [comment]
jenkins execute full downstream build

**Comment by nmirasch** [comment]
@jstastny-cz @cristianonicolai could you take a look? Thanks!

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenterTest.java [hunk 1]
```diff
@@ -15,290 +15,31 @@
...
-        verify(commentsService,
-               times(2)).getTaskComments(serverTemplateId,
-                                         containerId,
-                                         TASK_ID);
-        verify(view,
-               times(2)).redrawDataGrid();
-    }
-
-    @Test
-    public void emptyCommentNotAccepted() {
```
Comment (jstastny-cz) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenterTest.java]: do we have replacement for this?

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenter.java [hunk 2]
```diff
@@ -15,167 +15,71 @@
...
-        this.taskService = taskService;
-        this.identity = identity;
-    }
+    boolean sortAsc = false;
 
-    @PostConstruct
-    public void init() {
-        view.init(this);
-    }
+    int currentPage = 1;
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenter.java]: agree!

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenter.java [hunk 3]
```diff
@@ -15,167 +15,71 @@
...
-                            getTaskId(),
-                            commentId);
+    @Override
+    public TaskCommentsView getTaskCommentView() {
+        return view;
     }
 
-    public void addDataDisplay(final HasData<CommentSummary> display) {
-        dataProvider.addDataDisplay(display);
+    @Override
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenter.java]: updated

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsViewImpl.java [hunk 4]
```diff
@@ -15,276 +15,241 @@
...
+        onSortChange(sortAlphaAsc,
+                     sortAlphaDesc,
+                     false);
     }
 
-    @Override
-    public void newCommentsEnabled(final Boolean enabled) {
-        if (enabled) {
-            form.classList.remove("hidden");
+    private void addRemoveClassNameOnElement(HTMLElement element, boolean adding, String className) {
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsViewImpl.java]: done :)

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenterTest.java (hunk 1 above)
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/test/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsPresenterTest.java]: Now, that validation is covered at view test TaskCommentsViewImplTest with fieldValidationErrorTest and fieldValidationSuccessTest

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsWorkPresenter.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+import com.google.gwt.user.client.ui.IsWidget;
+import org.uberfire.client.mvp.UberView;
+
+@Dependent
+public class TaskCommentsWorkPresenter extends AbstractTaskCommentsPresenter {
+
+    private TaskWorkCommentsView workView;
+
+    boolean sortAsc = false;
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsWorkPresenter.java]: agree, updated!

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsViewImpl.java [hunk 6]
```diff
@@ -15,276 +15,241 @@
...
     }
 
     @Override
-    public void setErrorMessage(final String message) {
-        this.alert.classList.remove("hidden");
-        this.listContainer.classList.add("hidden");
-        this.message.textContent = message;
+    public void resetPagination() {
+        presenter.setCurrentPage(1);
+        onSortChange(sortAlphaAsc,
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsViewImpl.java]: updated

**Comment by nmirasch** [comment]
@jstastny-cz Thanks for your review, Could you take a look on changes? Thanks!

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        this.identity = identity;
+    }
+
+    public interface TaskCommentsView extends UberElement<AbstractTaskCommentsPresenter> {
+
+        void newCommentsHidden(Boolean hidden);
+
+        void clearCommentInputForm();
+
+        void setCommentList(final List<CommentSummary> commentList);
```
Comment (cristianonicolai) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java]: no need for final in interface.

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+        void resetPagination();
+
+        void hideLoadButton(boolean hidden);
+
+        void displayNotification(String text);
```
Comment (cristianonicolai) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java]: This is just an event to show the message, I would suggest keeping it in the presenter instead.

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+        void newCommentsHidden(Boolean hidden);
+
+        void clearCommentInputForm();
+
+        void setCommentList(final List<CommentSummary> commentList);
+
+        void resetPagination();
+
+        void hideLoadButton(boolean hidden);
```
Comment (cristianonicolai) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/AbstractTaskCommentsPresenter.java]: perhaps same approach, for hide/show load button

**Comment by nmirasch** [comment]
@cristianonicolai I have done almost all you suggested in this comments, but I'll need more time to re-think The simplification of the two, views two controllers. Let me give it a try

**Comment by nmirasch** [comment]
@cristianonicolai @jstastny-cz now there is only one view that can be configured to display header and just one presenter with the page size configurable. All your suggestions are included too. Please could you take another look on this? Thanks!

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsViewImpl.java [hunk 10]
```diff
@@ -15,276 +15,238 @@
...
-    @Override
-    public void clearCommentInput() {
-        newTaskCommentTextArea.setText("");
+    @Inject
+    @DataField("empty-list-item")
+    private Div emptyContainer;
+
+    protected TaskCommentsPresenter presenter;
+
+    protected native void tooltip(final HTMLElement e) /*-{
```
[the remark this answers is not in the document]
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskcomments/TaskCommentsViewImpl.java]: @cristianonicolai yes, absolutely right, updated!

**Comment by nmirasch** [comment]
Jenkins execute full downstream build

**Pull request merged**